libs/common/Memory/LimitedFixedBufferPool.cs

```diff
@@ -98,6 +108,23 @@ public unsafe PoolEntry Get(int size)
             return new PoolEntry(size, this);
         }
 
+        /// <summary>
+        /// Purge pool entries from all levels
+        /// </summary>
+        public void Purge()
+        {
+            for (var i = 0; i < numLevels; i++)
+            {
+                if (pool[i] == null) continue;
+                // Keep trying Dequeuing until no items left to free
+                while (pool[i].items.TryDequeue(out var entry))
+                {
+                    entry = null;
```

Comment:
What does this "= null" do, given that it immediately goes out of scope? And it is not done in Dispose().

 I also see a Thread.Yield() in Dispose() that is not here.
 
 Can Dispose() use Purge()?

Reply:
We cannot do Dispose on the entry because it will return it to the pool. 
Dequeuing it without setting it to null should be fine since it goes out of scope the GC will take care of it.